pkg/iterator/iter.go

```diff
@@ -191,9 +194,10 @@ func newIterator(args Args) *iterator {
 func (iter *iterator) run(ctx context.Context) error {
 	var err error
 
-	serviceAccount := iter.ServiceAccount // optional field
+	serviceAccount := iter.ServiceAccount                   // optional field
+	serviceAccountNamespace := iter.ServiceAccountNamespace // optional field
 	if serviceAccount == "" {
```

Comment:
The check for an empty name (only) is fine as long as `args.Validate` ensures both or none semantics.